editor/gui/scene_tree_editor.cpp

```diff
@@ -1208,7 +1208,10 @@ Variant SceneTreeEditor::get_drag_data_fw(const Point2 &p_point, Control *p_from
 		if (i < list_max) {
 			HBoxContainer *hb = memnew(HBoxContainer);
 			TextureRect *tf = memnew(TextureRect);
-			tf->set_texture(icons[i]);
+			int icon_size = get_theme_constant(SNAME("class_icon_size"), SNAME("Editor"));
+			Ref<Image> img = icons[i]->get_image();
+			img->resize(icon_size, icon_size);
+			tf->set_texture(ImageTexture::create_from_image(img));
```

Comment:
That's wrong. Resizing images is expensive and shouldn't be done. You need to enforce its maximum size. See https://github.com/godotengine/godot/pull/75472/commits/1522762dc986c130ad63cbf854514d730788a4cf.

You need to give the image a minimum size and then configure flags.

```suggestion
			tf->set_custom_minimum_size(Size2(icon_size, icon_size));
			tf->set_stretch_mode(TextureRect::STRETCH_KEEP_ASPECT_CENTERED);
			tf->set_expand_mode(TextureRect::EXPAND_IGNORE_SIZE);
			tf->set_texture(icons[i]);
```